### ISSUE TITLE
Extend CA1310 to warn about Dictionary<string, T> and HashSet<string>

### ISSUE THREAD

**Issue by verdie-g** [dotnet/roslyn-analyzers#6384]
### Analyzer

**Diagnostic ID**: [CA1310](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1310)

### Describe the improvement

In my company we found out that culture-sensitive comparison was responsible for massive performance issues. Using ordinal string comparison on our biggest service decreased its CPU by 20%. That's why we would like to get rid of implicit culture-sensitive comparison. The `CA1310` rule is helping but it doesn't warn about `Dictionary<string, T>` and `HashSet<string>` that use culture-sensitive comparison by default.

### Describe suggestions on how to achieve the rule

When a `Dictionary<string, T>` or a `HashSet<string>` is instantiated with no `IEqualityComparer` like the following:
```csharp
Dictionary<string, T> d = new();
HashSet<string> h = new();
```
 a warning would be emitted to specify it explicitly.
